light-client/tests/integration.rs

```diff
@@ -0,0 +1,121 @@
+//! Light Client integration tests.
+//!
+//! These are all ignored by default, since they test against running
+//! `tendermint node --proxy_app=kvstore`. They can be run using:
+//!
+//! ```
+//! cargo test -- --ignored
+//! ```
+
+use tendermint_light_client::{
+    components::{
+        clock::SystemClock,
+        io::{AtHeight, Io, IoError, ProdIo},
+        scheduler,
+        verifier::ProdVerifier,
+    },
+    evidence::{Evidence, EvidenceReporter},
+    fork_detector::ProdForkDetector,
+    light_client::{self, LightClient},
+    peer_list::PeerList,
+    state::State,
+    store::{memory::MemoryStore, LightStore},
+    supervisor::{Handle, Instance, Supervisor},
+    types::{PeerId, Status, Time, TrustThreshold},
+};
+
+use tendermint::abci::transaction::Hash as TransactionHash;
+
+use std::collections::HashMap;
+use std::time::Duration;
+
+fn make_instance(peer_id: PeerId, options: light_client::Options, io: ProdIo) -> Instance {
+    let trusted_state = io
+        .fetch_light_block(peer_id, AtHeight::Highest)
+        .expect("could not request latest light block");
+
+    let mut light_store = MemoryStore::new();
+    light_store.insert(trusted_state, Status::Trusted);
+
+    let state = State {
+        light_store: Box::new(light_store),
+        verification_trace: HashMap::new(),
+    };
+
+    let verifier = ProdVerifier::default();
+    let clock = SystemClock;
+    let scheduler = scheduler::basic_bisecting_schedule;
+
+    let light_client = LightClient::new(peer_id, options, clock, scheduler, verifier, io);
+
+    Instance::new(light_client, state)
+}
+
+struct TestEvidenceReporter;
+
+#[contracts::contract_trait]
+impl EvidenceReporter for TestEvidenceReporter {
+    fn report(&self, evidence: Evidence, peer: PeerId) -> Result<TransactionHash, IoError> {
+        panic!(
+            "unexpected fork detected for peer {} with evidence: {:?}",
+            peer, evidence
+        );
+    }
+}
+
+#[test]
+#[ignore]
+fn sync() {
+    let primary: PeerId = "BADFADAD0BEFEEDC0C0ADEADBEEFC0FFEEFACADE".parse().unwrap();
+    let witness: PeerId = "CEFEEDBADFADAD0C0CEEFACADE0ADEADBEEFC0FF".parse().unwrap();
+
+    let node_address: tendermint::net::Address = "tcp://127.0.0.1:26657".parse().unwrap();
+
+    let mut peer_map = HashMap::new();
+    peer_map.insert(primary, node_address.clone());
+    peer_map.insert(witness, node_address);
```

Comment:
Here the same RPC end-point is used twice. Do you think it makes sense add a comment that in a realistic scenario these would be different and that you'd check if the peerId matches?

Reply:
Addressed in 41283822e5a131ff1866fc174785b12b07d8af30.